crossbeam-channel/src/flavors/array.rs

```diff
@@ -410,6 +518,16 @@ impl<T> Channel<T> {
         }
     }
 
+    /// Attempts to receive messages without blocking.
+    pub fn try_recv_many<E: Extend<T>>(&self, buf: &mut E, limit: usize) -> Result<usize, TryRecvError> {
```

Comment:
Why the two seperate functions? (planning to reuse start_recv_many in future?)

Reply:
Yeah, I'm not sure there's a good reason to have it be separate.

I started out just trying to mimic the existing method structure, but @stjepang mentioned that the `select!` is already too complex for their liking, so it's not likely that this would ever have support added... I'll go ahead and collapse this down to just `try_recv_many`.

Reply:
Actually, I think it makes more sense maybe to just rename `start_recv_many` to `recv_many`.. my thought here is that I like being able to see `try_recv_many` as a separate function to show the logic of "if we got any items, we now have more free space, so notify the next waiting sender".

With everything compressed, those few lines of code get lost easily within the rest of the actual logic.

Reply:
@tobz then maybe the functions need different names. Just looking at the function names we have `recv_many` and `try_recv_many`, to me this says both functions do the same thing, but `try_recv_many` will return an error if it can't recveive anything, while the `recv_many` will block/panic. However this is not the case.

Maybe `pub(crate) fn try_recv_many` and `fn recv_many_internal`? To indicate you have to call `try_recv_many` to ensure senders are notified.

Reply:
Hmmm.... I think I can live with that, yeah. :)